### PR TITLE
Change sample test to read test arguments from yaml config file

### PR DESCRIPTION
**Pull request by numerology**
Part of #1750 
Now the params used for each sample test live in test/sample-test/configs dir. If no config file is provided, default param values will be used. Also, common constants used in sample test infra are moved into a dedicated python file.

Moving image injection and into config files guideline of adding a sample test will be in following PRs.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1987)

### CODE REVIEW

**Comment by gaoning777** [comment]
Could we add a readme to explain the steps to add a sample test?

**Comment by numerology** [comment]
> Could we add a readme to explain the steps to add a sample test?

Sure thing. Will do that in another PR.

File: test/sample-test/run_sample_test.py [hunk 1]
```diff
@@ -15,12 +15,10 @@
...
 import tarfile
 import utils
+import yaml
 from datetime import datetime
 from kfp import Client
-
-
-_TEST_TIMEOUT = 1200
-_XGB_TEST_TIMEOUT = 1800 # xgboost needs additional time to finish.
+from constants import XGB_TEST_TIMEOUT, TEST_TIMEOUT, CONFIG_DIR
```
[the remark this answers is not in the document]
Comment (numerology) [reply test/sample-test/run_sample_test.py]: Done.

File: test/sample-test/run_sample_test.py [hunk 2]
```diff
@@ -64,49 +62,42 @@ def check(self):
...
+      with open(DEFAULT_CONFIG, 'r') as f:
+        raw_args = yaml.safe_load(f)
+    except yaml.YAMLError as yamlerr:
+      raise RuntimeError('Illegal default config:{}'.format(yamlerr))
+    except OSError as ose:
+      raise FileExistsError('Default config not found:{}'.format(ose))
     else:
-      # Basic tests require no additional params.
-      params = {}
+      f.close()
```
Comment (gaoning777) [inline test/sample-test/run_sample_test.py]: I think with statement will automatically close the f, won't it?

File: test/sample-test/run_sample_test.py [hunk 3]
```diff
@@ -64,49 +62,32 @@ def check(self):
...
+    ###### Figure out test-specific arguments from associated config files. #######
+    test_args = {}
+    try:
+      with open(os.path.join(CONFIG_DIR, '%s.config.yaml' % self._testname), 'r') as f:
+          raw_args = yaml.safe_load(f)
+    except yaml.YAMLError as yamlerr:
+      print('No legit yaml config file found, use default args:{}'.format(yamlerr))
+    except OSError as ose:
+      print('Config file with the same name not found, use default args:{}'.format(ose))
     else:
```
[the remark this answers is not in the document]
Comment (gaoning777) [reply test/sample-test/run_sample_test.py]: Thanks. Learnt

**Comment by gaoning777** [comment]
> > Could we add a readme to explain the steps to add a sample test?
> 
> Sure thing. Will do that in another PR.

If not the readme, could you provide the spec about the config yaml in this PR? I think this should be a minimal work such as

testname:
    type: string
    required: true
test_timeout:
    type: float
    required: false
arguments:
    type: object
    required: false

This will be important because users need to know what the sample test infra is expecting from the configuration.
WDYT?

**Comment by numerology** [comment]
> > > Could we add a readme to explain the steps to add a sample test?
> > 
> > 
> > Sure thing. Will do that in another PR.
> 
> If not the readme, could you provide the spec about the config yaml in this PR? I think this should be a minimal work such as
> 
> ```
> testname:
>     type: string
>     required: true
> test_timeout:
>     type: float
>     required: false
> arguments:
>     type: object
>     required: false
> ```
> 
> This will be important because users need to know what the sample test infra is expecting from the configuration.
> WDYT?

Just to confirm. Are you talking about add this spec in a separated file, or inline in the comment in the config.yaml itself?

File: test/sample-test/run_sample_test.py (hunk 2 above)
Comment (numerology) [reply test/sample-test/run_sample_test.py]: Good catch. Fixed.

**Pull request merged**